### DISCUSSION TITLE
Error restart unsteady RANS 2nd order dual-time stepping

### DISCUSSION THREAD

**Discussion by rois1995** [su2code/SU2#1801] [Q&A]
**Describe the bug**
I am trying to restart an unsteady RANS 2nd order dual-time stepping simulation. I set READ_BINARY_RESTART=YES but it searches for the *.csv* restart file. This happens whenever I put SOLUTION_FILENAME equal to *name.dat* or just *name*. 

If *name_0000x.dat* exists in the folder then it crashes in CSolver::Restart_OldGeometry, saying that there is no *name_0000x.csv* . But if *name_0000x.dat* does not exist, then it fails in CSolver::Read_SU2_Restart_Binary, saying that there is no *name_0000(x-1).dat* (as expected).

I have uploaded the mesh, the config, and the restart files.

https://polimi365-my.sharepoint.com/:f:/g/personal/10507725_polimi_it/ErBtq0WVe45Eq_PFxKVQsFQBiYVWBoAoGURCxEBdkR3tAA?e=QODQkw

**Desktop (please complete the following information):**
 - OS: Linux [Ubuntu 20.04.5 LTS]
 - C++ compiler and version:  g++ (GCC) 9.4.0
 - MPI implementation and version: OpenMPI 4.0.3
 - SU2 Version: v7.4.0(develop)

**Comment by pcarruscag** [comment]
Unsteady problems with some types of deforming grids need to read the old position of the mesh because the grid velocity is computed with finite differences.
It happens that whoever coded this decided to read the old position from the CSV restart regardless of whether SU2 is reading binary or ascii restart files...
So in your case you need both .dat and .csv files to restart the solver :shrug: 

It's a bit of a mess and it could probably be done by reading the coordinates in the "normal restart". If you want to try doing something about it we can talk about it next Wednesday at 4pm CET https://meet.jit.si/SU2_DevMeeting

**Reply by rois1995** [reply]
Sure! I think we can manage it. See you on Wednesday!

**Reply by ArneVoss** [reply]
I just stumbled about the same problem. Is there a nicer solution, like using the binary restart only, yet?
